src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * @author leslieharland
+ */
+
+package duke.task;
+
+import duke.DukeException;
+import duke.command.Operation;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The Class TaskList.
+ */
+public class TaskList {
+    private List<Task> tasks;
+    private Ui ui;
+
+    /**
+     * Instantiates a new task list.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+        ui = new Ui();
+    }
+
+    /**
+     * Gets the tasks.
+     *
+     * @return the tasks
+     */
+    public List<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Gets the.
+     *
+     * @param v the v
+     * @return the task
+     */
+    public Task get(int v) {
+        return tasks.get(v);
+    }
+
+    /**
+     * Gets the size.
+     *
+     * @return the size
+     */
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Adds the task.
+     *
+     * @param t the t
+     */
+    public void addTask(Task t) {
+        tasks.add(t);
+    }
+
+    /**
+     * Adds the task.
+     *
+     * @param pos the pos
+     * @param t   the t
+     */
+    public void addTask(int pos, Task t) {
+        tasks.add(pos, t);
+    }
+
+    /**
+     * Adds the task.
+     *
+     * @param current the current
+     * @param storage the storage
+     * @throws DukeException the duke exception
+     */
+    public Task addTask(String[] current, Storage storage) throws DukeException {
+        assert (storage != null);
+        String[] words = Arrays.stream(current).skip(1).toArray(String[]::new);
+        String command = current[0];
+
+        if (words.length == 0) {
+            throw new DukeException("☹ OOPS!!! The description of a "
+                    + command + " cannot be empty.");
+        }
+        if (command.equalsIgnoreCase(Operation.DEADLINE.toString())) {
```

Comment:
It might be easier to view and for you to code if you try using a switch case method!